lib/http/request.js

```diff
@@ -189,18 +189,16 @@ const create = async function (method, path, options, hooks = {}) {
     return options.pathParams[args[1]] || args[0]
   })
 
-  const hasQueryParams = options.queryParams ? Object.keys(options.queryParams).length > 0 : false
-
-  if (hasQueryParams && path.includes('?')) {
-    const parts = path.split('?')
-    options.queryParams = {
-      ...QueryString.parse(parts[1]),
-      ...options.queryParams
+  if (options.queryParams) {
+    if (path.includes('?')) {
+      let queryStr;
+      [path, queryStr] = path.split('?', 2)
+      options.queryParams = {
+        ...QueryString.parse(queryStr),
+        ...options.queryParams
+      }
     }
-    path = parts[0]
-  }
 
-  if (hasQueryParams) {
```

Comment:
Combined into the above `hasQueryParams` if statement (`if (options.queryParams) {`)